### ISSUE TITLE
[BUG] While defining UDT local variable name collide with name from external namespace and generates `.value()` calls on types

### ISSUE THREAD

**Issue by filipsajdak** [hsutter/cppfront#272]
The current implementation of cppfront (fb4a9fc4bf43ec17e0c9d6c863221032c344677f) process correctly the following code:
```cpp
namespace ns {
    struct config {};
}

x : type = {
    config: ns::config;
    c1: ns::config;
    public c2: ns::config;
    protected c3: ns::config;
    private c4: ns::config;
}
```
But generates bad cpp1 code:
```cpp
class x   {
    private: ns::config.value() config; 
    private: ns::config.value() c1; 
    public: ns::config.value() c2; 
    protected: ns::config.value() c3; 
    private: ns::config c4; 
};
```

If you remove the definition of `config` member:
```cpp
namespace ns {
    struct config {};
}

x : type = {
    // config: ns::config;
    c1: ns::config;
    public c2: ns::config;
    protected c3: ns::config;
    private c4: ns::config;
}
```
Everything works as expected:
```cpp
class x   {
    // config: ns::config;
    private: ns::config c1; 
    public: ns::config c2; 
    protected: ns::config c3; 
    private: ns::config c4; 
};
```

### Notes

`config` can be defined using the cpp1 or cpp2 way. It looks like a collision between a local name and a global name defined in an external namespace.

**Comment by hsutter** [comment]
I could see why it might have that bug, and was already thinking of how to fix it, but when I tried to reproduce the error it no longer manifests for me. I think I recently did touch the code that adds `.value()` to fix something else, so it's possible it fixed this too.

Can you still repro this bug? If not, we can close this until we encounter it again.

**Comment by filipsajdak** [comment]
Correct. Fixed in commit 8ed2e72d2b8916b3b699f2cffe15f7b4ed72f528
